Урок 7. Практическое задание/task_2/task_2_1.py

```diff
@@ -15,3 +15,93 @@
 
 сделайте замеры на массивах длиной 10, 100, 1000 элементов
 """
+
+
+from random import randint
+from timeit import timeit
+
+
+# Гномья сортировка
+def gnome_sort(lst_obj):
+    len_list = len(lst_obj)
+    i = 0
+    while i < len_list:
+        if i == 0:
+            i += 1
+        if lst_obj[i] >= lst_obj[i - 1]:
+            i += 1
+        else:
+            lst_obj[i], lst_obj[i - 1] = lst_obj[i - 1], lst_obj[i]
+            i -= 1
+    return lst_obj[(len(lst_obj) // 2)]
+
+
+# Сортировка расчёской
+def brush_sort(lst_obj):
+    step = len(lst_obj) - 1
+    while step > 0:
+        for i in range(0, len(lst_obj)-step):
+            if lst_obj[i] > lst_obj[i+step]:
+                lst_obj[i], lst_obj[i+step] = lst_obj[i+step], lst_obj[i]
+        step = int(step // 1.25)
+    return lst_obj[(len(lst_obj) // 2)]
+
+
+# Сортировка Шелла
+def shell_sort(lst_obj):
+    step = len(lst_obj)//2
+    while step > 0:
+        for i in range(step, len(lst_obj)):
+            delta = i - step
+            while delta >= 0 and lst_obj[delta] > lst_obj[i]:
+                lst_obj[delta], lst_obj[i] = lst_obj[i], lst_obj[delta]
+                delta -= step
+        step //= 2
+    return lst_obj[(len(lst_obj) // 2)]
+
+
+orig_list = [randint(-100, 100) for _ in range(0, 2 * 10 + 1)]
+orig_list_1 = [randint(-100, 100) for _ in range(0, 2 * 100 + 1)]
+orig_list_2 = [randint(-100, 100) for _ in range(0, 2 * 1000 + 1)]
+
```

Comment:
orig_list = [randint(-100, 100) for _ in range(0, 2 * 10 + 1)]
orig_list_1 = [randint(-100, 100) for _ in range(0, 2 * 100 + 1)]
orig_list_2 = [randint(-100, 100) for _ in range(0, 2 * 1000 + 1)]

2m+1